### DISCUSSION TITLE
dotnet cake and CakeExecuteScript tries to run Cake.exe

### DISCUSSION THREAD

**Discussion by mattleibow** [cake-build/cake#3454] [Q&A]
I am switching to using a .NET local tool, and when I try invoke the `CakeExecuteScript` alias, it fails:

```
Cake: Process returned an error (exit code 255).
```

When I do a verbose run, I see it is trying to use .NET Framework on Linux:

```
Could not resolve path for tool "mono" using these directories: /usr/local/sbin,/usr/local/bin,/usr/sbin,/usr/bin,/sbin,/bin
/work/tools/Cake/Cake.exe is a .NET Framework executable, you might need to install Mono for it to execute successfully.
Executing: /work/tools/Cake/Cake.exe "/work/native/linux/build.cake" --target="Default" --configuration="Release" --buildarch="x64" --variant="alpine"
An error occurred when executing task 'externals-linux'.
Error: System.AggregateException: One or more errors occurred. (Cake: Process returned an error (exit code 255).)
 ---> Cake.Core.CakeException: Cake: Process returned an error (exit code 255).
```

Is this a supported scenario?

What is the correct/better way?

**Comment by nils-a** [comment]
@mattleibow `CakeExecuteScript` tries to find a runner for Cake files, then uses this runner. 
It searches for `Cake.exe`, `cake` and `cake.dll` [in that order](https://github.com/cake-build/cake/blob/a1893a2bba26f49e089d66beeb78421cedf6126d/src/Cake.Common/Tools/Cake/CakeRunner.cs#L182).
In your case it seems it finds the "old" `cake.exe` under `/work/tools/cake/cake.exe`. 

Is it possible that your `tools` folder contains a version of the Cake runner for .NET Framework (probably from a time from "before" your switch to the .NET tool)?

**Reply by mattleibow** [reply]
I think that was the issue. If I am running as a .net global tool, none of those exist.

I'll check again to see, but if I have no mono or no tools directory, the only way to run is with 'dotnet cake'

**Reply by nils-a** [reply]
yes, and running `dotnet cake` is a good choice. But you need to make sure that non-windows systems that have no mono installed don't have the Cake runner for .NET Framework (i.e. `cake.exe`) available. (For example, don't place `./tools/Cake` under version control.)
